testsuite/forge/src/backend/k8s_deployer/deployer.rs

```diff
@@ -0,0 +1,444 @@
+/// There are global values such as profile, era, and namespace as well as component-specific values
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ForgeDeployerValues {
+    pub profile: String,
+    pub era: String,
+    pub namespace: String,
+
+    // component specific values
+    // TODO: add an options reference. Ideally this customization is almost always optional and instead handled by the profiles
+    pub indexer_grpc_values: Option<serde_json::Value>,
+    pub indexer_processor_values: Option<serde_json::Value>,
+}
+
+/// The ForgeDeployerManager is responsible for managing the lifecycle of forge deployers, wihch deploy the
+/// forge components to the k8s cluster.
+pub struct ForgeDeployerManager {
+    // all the k8s APIs we need. Specifying each API separately allows for easier testing
+    pub jobs_api: Arc<dyn ReadWrite<Job>>,
+    pub config_maps_api: Arc<dyn ReadWrite<ConfigMap>>,
+    pub namespace_api: Arc<dyn ReadWrite<Namespace>>,
+    pub serviceaccount_api: Arc<dyn ReadWrite<ServiceAccount>>,
+    pub rolebinding_api: Arc<dyn ReadWrite<RoleBinding>>,
+
+    // the values to use for the deployer, including namespace, era, etc
+    pub values: ForgeDeployerValues,
+}
+
+#[derive(Clone, Copy)]
+pub enum ForgeDeployerType {
+    Indexer,
+    Testnet,
+}
+
+impl fmt::Display for ForgeDeployerType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ForgeDeployerType::Indexer => write!(f, "indexer"),
+            ForgeDeployerType::Testnet => write!(f, "testnet"),
+        }
+    }
+}
+
+impl ForgeDeployerManager {
+    pub fn from_k8s_client(kube_client: kube::Client, values: ForgeDeployerValues) -> Self {
+        let jobs_api = Arc::new(K8sApi::from_client(
+            kube_client.clone(),
+            Some(values.namespace.clone()),
+        ));
+        let config_maps_api = Arc::new(K8sApi::from_client(
+            kube_client.clone(),
+            Some(values.namespace.clone()),
+        ));
+        let namespace_api = Arc::new(K8sApi::from_client(kube_client.clone(), None));
+        let serviceaccount_api = Arc::new(K8sApi::from_client(
+            kube_client.clone(),
+            Some(values.namespace.clone()),
+        ));
+        let rolebinding_api = Arc::new(K8sApi::from_client(
+            kube_client.clone(),
+            Some(values.namespace.clone()),
+        ));
+
+        // ensure it lives long enough between async
+        Self {
+            jobs_api,
+            config_maps_api,
+            namespace_api,
+            serviceaccount_api,
+            rolebinding_api,
+            values,
+        }
+    }
+
+    /// Given a deployer type return the name to use for k8s components
+    /// This is the canonical name for the deployer and each of its components
+    pub(crate) fn get_name(&self, deployer_type: ForgeDeployerType) -> String {
```

Comment:
Lets pass name from the toplevel instead of generating a name here

I also dont think we need type or era so that would also simplify this logic removing two other concepts that can either be passed transparently to the backend or used elsewhere